sympy/solvers/tests/test_solveset.py

```diff
@@ -1086,3 +1071,66 @@ def test_issue_8715():
         (Interval.open(-2, oo) - FiniteSet(0))
     assert solveset(eq.subs(x,log(x)), x, S.Reals) == \
         Interval.open(exp(-2), oo) - FiniteSet(1)
+
+def test_issue_9824():
+    assert solveset(sin(x)**2 - 2*sin(x) + 1, x, domain=S.Reals) ==\
+     ImageSet(Lambda(n, 2*n*pi + pi/2), S.Integers)
+
+
+def test_issue_9531_and_9606():
+    # complex solution
+    assert solveset(sinh(x)) == ImageSet(Lambda(n, n*I*pi), S.Integers)
+    # real solution
+    assert solveset(sinh(x), x, S.Reals) == FiniteSet(0)
+
+
+def test_issue_7914():
+    assert solveset(sin(2*x)*cos(x) + cos(2*x)*sin(x) -1, x) ==\
+    ImageSet(Lambda(n, pi*(4*n - 3)/6), S.Integers) + ImageSet(Lambda(n, 2*n*pi - pi/2), S.Integers)
+
+
+def test_issue_10671():
+    assert solveset(sin(y), y, Interval(0, pi)) == \
+    Intersection(Interval(0, pi), ImageSet(Lambda(n, n*pi), S.Integers))
+
+
+def test_simplifed_trig_solution():
+    assert solveset(cos(x) + cos(3*x) + cos(5*x), x, S.Reals) == \
+    ImageSet(Lambda(n, n*pi/6), S.Integers)
+    assert solveset(sin(x)**2 -sin(x) -2, x, S.Reals) == \
+    ImageSet(Lambda(n, 2*n*pi + 3*pi/2), S.Integers)
+    assert solveset(sin(x+pi/4) -1/sqrt(2), x, S.Reals) == \
+    ImageSet(Lambda(n, (-1)**n*pi/4 + n*pi - pi/4), S.Integers)
+    assert solveset(cos(x)**2 +cos(x) -sin(x)**2, x, S.Reals) == \
+    ImageSet(Lambda(n, pi*(2*n - 1)/3), S.Integers) +\
+     ImageSet(Lambda(n, 2*n*pi + 5*pi/3), S.Integers)
+    assert solveset((sin(x)+1)**2, x, S.Reals) == \
+    ImageSet(Lambda(n, 2*n*pi + 3*pi/2), S.Integers)
+
+
+@SKIP("Comparison error")
+def test_simplifed_trig_solution_1():
+    # `factor_list` is helpful for these types of case.
+    soln = ImageSet(Lambda(n, -pi*(4*n - 3)/6), S.Integers) +\
+     ImageSet(Lambda(n, -pi*(2*n - 1)/4), S.Integers) + \
+     ImageSet(Lambda(n, pi*(2*n - 1)/4), S.Integers)
+    assert solveset(4*sin(x)**3  + 2*sin(x)**2 - 2*sin(x) - 1, x) == \
+    soln
+
+    s = solveset((tan(x)**2 - 3).rewrite(exp), x)
+    assert s == ImageSet(Lambda(n, 2*n*pi + 4*pi/3), \
+        S.Integers) + ImageSet(Lambda(n, 2*n*pi + 2*pi/3), S.Integers) \
+    + ImageSet(Lambda(n, 2*n*pi + 5*pi/3), S.Integers) + \
+    ImageSet(Lambda(n, 2*n*pi + pi/3), S.Integers)
+
+
+@XFAIL
+def test_simplifed_trig_solution_2():
```

Comment:
There may be some cases where we need good way to separate 2-3 imageset to make it simple. Right now for these cases it is returning original solution.